manifests/init.pp

```diff
@@ -25,39 +25,37 @@
 # @param event_logs_merge [Boolean] Whether $event_logs should merge all hiera sources, or use simple automatic parameter lookup
-
+#
 class winlogbeat (
-  $major_version        = undef,
-  $package_ensure       = $winlogbeat::params::package_ensure,
-  $service_ensure       = $winlogbeat::params::service_ensure,
-  $service_enable       = $winlogbeat::params::service_enable,
-  $service_provider     = $winlogbeat::params::service_provider,
-  $registry_file        = $winlogbeat::params::registry_file,
-  $config_file          = $winlogbeat::params::config_file,
-  $outputs              = $winlogbeat::params::outputs,
-  $shipper              = $winlogbeat::params::shipper,
-  $logging              = $winlogbeat::params::logging,
-  $run_options          = $winlogbeat::params::run_options,
-  $conf_template        = undef,
-  $download_url         = undef,
-  $install_dir          = $winlogbeat::params::install_dir,
-  $tmp_dir              = $winlogbeat::params::tmp_dir,
+  Optional[String]          $major_version        = undef,
+  String                    $package_ensure       = $winlogbeat::params::package_ensure,
+  String                    $service_ensure       = $winlogbeat::params::service_ensure,
+  Boolean                   $service_enable       = $winlogbeat::params::service_enable,
+  Optional[String]          $service_provider     = $winlogbeat::params::service_provider,
+  String                    $registry_file        = $winlogbeat::params::registry_file,
+  Stdlib::Windowspath       $config_file          = $winlogbeat::params::config_file,
+  Hash                      $outputs              = $winlogbeat::params::outputs,
+  Hash                      $shipper              = $winlogbeat::params::shipper,
+  Hash                      $logging              = $winlogbeat::params::logging,
+  Hash                      $run_options          = $winlogbeat::params::run_options,
+  Optional[String]          $conf_template        = undef,
+  Optional[Stdlib::HTTPUrl] $download_url         = undef,
+  String                    $install_dir          = $winlogbeat::params::install_dir,
+  String                    $tmp_dir              = $winlogbeat::params::tmp_dir,
   #### v5 only ####
-  $use_generic_template = $winlogbeat::params::use_generic_template,
-  $beat_name            = $winlogbeat::params::beat_name,
-  $tags                 = $winlogbeat::params::tags,
-  $queue_size           = $winlogbeat::params::queue_size,
-  $max_procs            = $winlogbeat::params::max_procs,
-  $fields               = $winlogbeat::params::fields,
-  $fields_under_root    = $winlogbeat::params::fields_under_root,
-  $metrics              = undef,
+  Boolean                   $use_generic_template = $winlogbeat::params::use_generic_template,
+  Stdlib::Fqdn              $beat_name            = $winlogbeat::params::beat_name,
+  Array                     $tags                 = $winlogbeat::params::tags,
```

Comment:
It's always good if we can qualify this more precisely, e.g. `Array[String[1]]` for an array of non-emty strings.